packages/graphql/src/schema/validation/custom-rules/utils/path-parser.ts

```diff
@@ -33,21 +34,34 @@ export type ObjectOrInterfaceWithExtensions =
 
 export function getPathToNode(
     path: readonly (number | string)[],
-    ancenstors: readonly (ASTNode | readonly ASTNode[])[]
+    ancestors: readonly (ASTNode | readonly ASTNode[])[]
 ): [
     Array<string>,
     ObjectOrInterfaceWithExtensions | FieldDefinitionNode | undefined,
     ObjectOrInterfaceWithExtensions | undefined
 ] {
-    const documentASTNodes = ancenstors[1];
-    if (!documentASTNodes || (Array.isArray(documentASTNodes) && !documentASTNodes.length)) {
+    if (!ancestors || !ancestors[0]) {
         return [[], undefined, undefined];
     }
-    const [, definitionIdx] = path;
-    const traversedDefinition = documentASTNodes[definitionIdx as number];
+    let traversedDefinition, pathIdx;
+    // if visiting from the document level
+    if (!Array.isArray(ancestors[0]) && (ancestors[0] as ASTNode).kind === Kind.DOCUMENT) {
```

Comment:
Maybe make that comment explicit?

```suggestion
    const visitingFromDocumentLevel = !Array.isArray(ancestors[0]) && (ancestors[0] as ASTNode).kind === Kind.DOCUMENT
    if (visitingFromDocumentLevel) {
```